backend/app/core/rag/vector_store.py

```
@@ -1,42 +1,177 @@
+            chunks = json.load(file)
+
+        index = faiss.read_index(str(index_file))
+        cluster_kb: Dict[str, Any] = {
+            "chunks": chunks,
+            "faiss_index": index,
+            "dim": index.d,
+        }
+
+        if corpus_file.exists():
+            with open(corpus_file, "rb") as file:
+                corpus = pickle.load(file)
```

Ruff (0.15.12)

[error] `pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)